packages/connector/src/provider.ts

```diff
@@ -55,8 +53,15 @@ export type GianoProviderInjection = {
   onCredentialSignedIn(credential: PublicKeyCredential): Promise<boolean>; // method to control if the credential is signed in or not
   getPublicKeyByCredentialId(rawId: ArrayBuffer): Promise<{ x: Hex; y: Hex }>;
   onCredentialKey(rawId: ArrayBuffer, xyVector: { x: Hex; y: Hex }): Promise<void>;
-  /** @deprecated */
-  onUserOperationSigned?: (signedUserOp: any) => Promise<any>; // optional hook for backend validation and submission
+  /**
+   * Override for sending user operations manually.
+   * When provided, this function will handle the submission of signed user operations
+   * instead of sending them directly to the bundler.
+   *
+   * @param signedUserOp - The complete signed user operation ready for submission
+   * @returns Promise that resolves to the transaction receipt
+   */
+  submitUserOperation?: (signedUserOp: any) => Promise<any>;
```

Comment:
This should be properly typed, it should also return the user operation hash, not the transaction awaited receipt, so the frontend can do the waiting separate from the call